diff --git a/apps/frontend/src/router.ts b/apps/frontend/src/router.ts
index 7b27e16b..edc922ef 100644
--- a/apps/frontend/src/router.ts
+++ b/apps/frontend/src/router.ts
@@ -15,6 +15,7 @@ import Router from 'vue-router';
 Vue.use(Router);
 
 const router = new Router({
+  base: "#####ROUTER_BASE_PATH#####",
   mode: 'history',
   routes: [
     {
diff --git a/apps/frontend/src/store/server.ts b/apps/frontend/src/store/server.ts
index b81f7371..100f048e 100644
--- a/apps/frontend/src/store/server.ts
+++ b/apps/frontend/src/store/server.ts
@@ -84,7 +84,7 @@ class Server extends VuexModule implements IServerState {
   SET_TOKEN(newToken: string) {
     this.token = newToken;
     localToken.set(newToken);
-    axios.defaults.headers.common['Authorization'] = `Bearer ${newToken}`;
+    axios.defaults.headers.common['X-Heimdall-Authorization'] = `Bearer ${newToken}`;
   }
 
   @Mutation
@@ -144,6 +144,7 @@ class Server extends VuexModule implements IServerState {
    */
   @Action
   public async CheckForServer() {
+    axios.defaults.baseURL = "#####AXIOS_BASE_URL#####";
     // This is the only function that manipulates the loading state. If loading is already set
     // then we have already loaded the server information and there is no need to check again.
     if (!this.loading) {
diff --git a/cmd.sh b/cmd.sh
index 74db5f94..05cdfa62 100644
--- a/cmd.sh
+++ b/cmd.sh
@@ -1,4 +1,59 @@
 #!/bin/sh
+
+if [[ "$HEIMDALL_NAMESPACE" == "" ]]; then
+  HEIMDALL_NAMESPACE="cattle-compliance-system"
+fi
+
+if [[ "$HEIMDALL_NGINX_WORKLOAD" == "" ]]; then
+  HEIMDALL_NGINX_WORKLOAD="heimdall-nginx"
+fi
+
+if [[ "$HEIMDALL_NGINX_PORT" == "" ]]; then
+  HEIMDALL_NGINX_PORT="80"
+fi
+
+if [[ "$RCIDF_PATH" == "" ]]; then
+  RCIDF_PATH="/rancher/id"
+fi
+
+if [ ! -f "$RCIDF_PATH" ]; then
+  if [[ "$PUBLIC_PATH" == "" ]]; then
+    echo "One of $RCIDF_PATH or PUBLIC_PATH must be defined."
+    exit 1
+  fi
+else
+  # RCIDF PATH exists
+  # use it to construct PUBIC_PATH
+  CLUSTERID=$(cat $RCIDF_PATH)
+  PUBLIC_PATH="/k8s/clusters/$CLUSTERID/api/v1/namespaces/$HEIMDALL_NAMESPACE/services/http:$HEIMDALL_NGINX_WORKLOAD:$HEIMDALL_NGINX_PORT/proxy/"
+fi
+
+if [[ "$VUE_PUBLIC_PATH" == "" ]]; then
+  VUE_PUBLIC_PATH=$PUBLIC_PATH
+fi
+
+if [[ "$ROUTER_BASE_PATH" == "" ]]; then
+  ROUTER_BASE_PATH=$PUBLIC_PATH
+fi
+
+if [[ "$AXIOS_BASE_URL" == "" ]]; then
+  AXIOS_BASE_URL=$PUBLIC_PATH
+fi
+
+sed -i "s|#####PUBLIC_PATH#####|$PUBLIC_PATH|g;" /app/dist/frontend/js/*
+
+sed -i "s|#####VUE_PUBLIC_PATH#####|$VUE_PUBLIC_PATH|g;" /app/dist/frontend/js/*
+sed -i "s|#####VUE_PUBLIC_PATH#####|$VUE_PUBLIC_PATH|g;" /app/dist/frontend/*.html
+
+sed -i "s|#####ROUTER_BASE_PATH#####|$ROUTER_BASE_PATH|g;" /app/dist/frontend/js/*
+sed -i "s|#####AXIOS_BASE_URL#####|$AXIOS_BASE_URL|g;" /app/dist/frontend/js/*
+
+echo "Configured with... "
+echo "PUBLIC_PATH=$PUBLIC_PATH"
+echo "VUE_PUBLIC_PATH=$VUE_PUBLIC_PATH"
+echo "ROUTER_BASE_PATH=$ROUTER_BASE_PATH"
+echo "AXIOS_BASE_URL=$AXIOS_BASE_URL"
+
 set -e
 yarn backend sequelize-cli db:migrate
 yarn backend sequelize-cli db:seed:all
